pjnath/src/pjnath/ice_strans.c

```diff
@@ -1016,8 +1024,10 @@ static void destroy_ice_st(pj_ice_strans *ice_st)
 
     /* Destroy ICE if we have ICE */
     if (ice_st->ice) {
-	pj_ice_sess_destroy(ice_st->ice);
+	ice_st->ice_prev = ice_st->ice;
 	ice_st->ice = NULL;
+	pj_ice_sess_detach_grp_lock(ice_st->ice_prev, &ice_st->ice_prev_hndlr);
+	pj_ice_sess_destroy(ice_st->ice_prev);
     }
```

Comment:
I think this block can be left unchanged.

Reply:
Setting `ice_st->ice` to `NULL` before calling `pj_ice_sess_destroy()` is important to avoid stray crash (IIRC, ICE strans uses `ice_st->ice != NULL` checks in many parts)?

Reply:
ah, I phrased my comment wrongly.
what I meant was that we don't seem to need to detach the group lock here and can safely rely on the group lock handler's mechanism.

Reply:
ah okay, this code is piggybacking `ice_prev` clean-up [here](https://github.com/pjsip/pjproject/pull/2903/files#diff-45e0dd75b3d3df135b497d029a81f78fae95e827c76a7e34c9b0ce9a51c4b0f3R990), it works, but it may not look clean.